### PR TITLE
Add base Sentry integration

### LINKED ISSUE #635: Add Sentry to app for crash reporting and troubleshooting
Requested access: https://bugzilla.mozilla.org/show_bug.cgi?id=1485740

Expecting to be set up like Firefox and Focus iOS apps.

### PR DESCRIPTION
**Pull request by joeyg**
Resolves #635. Let me know if this code should be put somewhere else in the app.

Needs: DSN added to Info.plist, tests.

### CODE REVIEW

**Comment by devinreams** [comment]
Thank you, @joeyg 😄

**Comment by devinreams** [comment]
To attempt to match what the other projects have done, I just added our public/private keys (DSN) to buddybuild as an environment variable: `SENTRY_DSN`.

From here it looks like we need two more things to link that up?

1. Add a reference in the `info.plist`
  https://github.com/mozilla-mobile/firefox-ios/blob/ba6e61e1d9afba8c710f4de67e0e37e42392529b/Client/Info.plist#L165-L166

2. Update the `buddybuild_prebuild` script to get/set that value
  https://github.com/mozilla-mobile/firefox-ios/blob/ba6e61e1d9afba8c710f4de67e0e37e42392529b/buddybuild_prebuild.sh#L40-L50

Iis my quick look at `firefox-ios` correct?

File: buddybuild_postbuild.sh [hunk 1]
```diff
@@ -1,2 +1,4 @@
 #!/usr/bin/env bash
 bash <(curl -s https://codecov.io/bash) -J Lockbox -t $CODECOV_TOKEN
+
+/usr/libexec/PlistBuddy -c "Set SentryDSN $SENTRY_DSN" "lockbox-ios/Common/Resources/Info.plist"
```
[the remark this answers is not in the document]
Comment (devinreams) [reply buddybuild_postbuild.sh]: Oh duh, I wasn't reading close enough. Yeah, I'll move it since it needs to be set before the build steps happen. Thanks

**Comment by sashei** [comment]
@devinreams specs look good, just needs the actual injection part!!

**Comment by devinreams** [comment]
Sorry, I should have included this with my last comment. Thanks ya'll. 😊 

ihave

**Comment by devinreams** [comment]
I tested a forced crash on the "Lock Now" button and this seems to be working!

https://sentry.prod.mozaws.net/operations/lockbox-ios/issues/4608786/

@sashei good to merge? 🙏

**Pull request merged**